### PR TITLE
Fix issue #765

### PR DESCRIPTION
**Pull request by Hannnnnn**
Fix issue: https://github.com/remkop/picocli/issues/765
I add usageSplit property for Option and Parameters.
If the user want to provide a user-friendly usage information without escaped char in regex.
He or she could add usageSplit in  Option and Parameters.
If it is not set by user, usageSpitRegex will be the same as splitRegex in class Argspec and class Builder. So it does not affect the orginal tests.

### CODE REVIEW

File: src/main/java/picocli/CommandLine.java [hunk 1]
```diff
@@ -8166,6 +8177,7 @@ private static String describe(ArgSpec argSpec, String separator, String value)
                     description = option.description();
                     descriptionKey = option.descriptionKey();
                     splitRegex = option.split();
+                    usageSplitRegex = option.usageSplit().equals("") ? splitRegex : option.usageSplit();
```
Comment (NewbieOrange) [inline src/main/java/picocli/CommandLine.java:8180]: Why not isEmpty()?

File: src/main/java/picocli/CommandLine.java [hunk 2]
```diff
@@ -8166,6 +8177,7 @@ private static String describe(ArgSpec argSpec, String separator, String value)
                     description = option.description();
                     descriptionKey = option.descriptionKey();
                     splitRegex = option.split();
+                    usageSplitRegex = option.usageSplit().isEmpty() ? splitRegex : option.usageSplit();
```
Comment (remkop) [inline src/main/java/picocli/CommandLine.java:8180]: Let's just store the annotation value (and we can decide later whether to use `splitRegex` or `usageSplitRegex`):

```
usageSplitRegex = option.usageSplit();
```

File: src/main/java/picocli/CommandLine.java [hunk 3]
```diff
@@ -8197,6 +8209,7 @@ private static String describe(ArgSpec argSpec, String separator, String value)
                         description = parameters.description();
                         descriptionKey = parameters.descriptionKey();
                         splitRegex = parameters.split();
+                        usageSplitRegex = parameters.usageSplit().equals("") ? splitRegex : parameters.usageSplit();
```
Comment (remkop) [inline src/main/java/picocli/CommandLine.java:8212]: Let's just store the annotation value (and we can decide later whether to use `splitRegex` or `usageSplitRegex`):

```
usageSplitRegex = parameters.usageSplit();
```

File: src/main/java/picocli/CommandLine.java [hunk 4]
```diff
@@ -8370,7 +8383,12 @@ private static String inferLabel(String label, String fieldName, ITypeInfo typeI
                 public T converters(ITypeConverter<?>... cs) { this.converters = Assert.notNull(cs, "type converters").clone(); return self(); }
 
                 /** Sets a regular expression to split option parameter values or {@code ""} if the value should not be split, and returns this builder. */
-                public T splitRegex(String splitRegex)       { this.splitRegex = Assert.notNull(splitRegex, "splitRegex"); return self(); }
```
Comment (remkop) [inline src/main/java/picocli/CommandLine.java:8373]: Let's leave the existing `splitRegex(String)` setter as it is, and add a separate new setter `ArgSpec.Builder::usageSplitRegex(String)`.

We also need new getters `String ArgSpec.Builder::usageSplitRegex()` and  `String ArgSpec::usageSplitRegex()`.

File: src/main/java/picocli/CommandLine.java [hunk 5]
```diff
@@ -14510,7 +14528,7 @@ public Text renderParameterLabel(ArgSpec argSpec, Ansi ansi, List<IStyle> styles
                 if (argSpec.hideParamSyntax()) { return colorScheme.apply((argSpec.isOption() ? separator() : "") + argSpec.paramLabel(), styles); }
 
                 Text paramName = colorScheme.apply(argSpec.paramLabel(), styles);
-                String split = argSpec.splitRegex();
+                String split = argSpec.usageSplitRegex();
```
Comment (remkop) [inline src/main/java/picocli/CommandLine.java:14531]: This is where we choose whether to use usageSplitRegex or splitRegex:

```java
String split = isEmpty(argSpec.usageSplitRegex()) ? argSpec.splitRegex() : argSpec.usageSplitRegex();
```

File: src/test/java/picocli/test/UsageSplitExecution.java [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+      public void run() {
+
+      }
+    }
+
+    public static void main (String []args) {
+      CommandLine commandLine = new CommandLine(new UsageSplitCommand());
+      commandLine.usage(commandLine.getOut());
+    }
+}
```
Comment (remkop) [inline src/test/java/picocli/test/UsageSplitExecution.java:24]: Let's make this a JUnit test:

```java
package picocli;

import org.junit.Test;
import picocli.CommandLine.Command;
import picocli.CommandLine.Option;

import static org.junit.Assert.assertEquals;
import static picocli.CommandLine.Help.Ansi.OFF;

public class UsageSplitExecutionTest {
    @Test
    public void testSimple() {
        @Command(name = "UsageSplit")
        class Simple {
            @Option(names = "v", split = "\\|", usageSplit = "|")
            String args[] = {};
        }

        String expected = String.format("" +
                "Usage: UsageSplit [v=<args>[|<args>...]]...%n" +
                "      v=<args>[|<args>...]%n" +
                "%n");
        String actual = new CommandLine(new Simple()).getUsageMessage(OFF);
        assertEquals(expected, actual);
    }
    
    // more tests for using ArgSpec.Builder usageSplit getter and setter
}
```

**Comment by Hannnnnn** [comment]
Thank you for reply.
I have followed you advices, change the code and add some tests.

**Comment by remkop** [comment]
Merged into master. 
Thank you for helping making picocli a better library!

**Pull request merged**